// viaden-modules/config/css-selector.js
(function () {

	"use strict";
	/*global */

	var selector;

	selector = {

		register: {

			registrationForm: '.registration-form', // div
			eMail: '#email', // text
			phoneNumber: '#cell', // number
			userName: '#userName', // text
			password: '#password', // text
			confirmPassword: '#confirm-password', // text
			title: '#title option:nth-child(2)', // select
			firstName: '#firstname', // text
			lastName: '#lastname', // text
			birthDate: '#birthdate', // text
			country: '#cor option[value=GB]', // select
			postcode: '#postcode', // text
			postcodeFind: '#postcodeButton', // button
			postcodeSuggestion: '#postcodeSuggestionSelect option[value="0"]', // select
			submit: 'input[type="submit"][name="submit"]', // button
			termsContent: '.accept-terms-popup .scroll-box.info-content', // div
			termsSubmit: '.accept-terms-popup [data-action="accept"]', // button
			deposit: '.deposit-btn' // button
		},

		joinNow: '.join-now-btn'







	};



	module.exports = selector;

}());
