// viaden-modules/util/index.js
(function () {

	"use strict";
	/*global */

	var util, fs, webDriver, mainCfg, path, startPath;

	fs = require('fs');

	webDriver = require('selenium-webdriver');

	mainCfg = require('viaden-modules/config/main.js');

	path = require('path');

	startPath = process.cwd();

	util = {

		keyList: ['isMobile', 'sendMail'],

		inKeyList: function(field) {
			return this.keyList.indexOf(field) !== -1;
		},

		createWebDriverClient: function() {

			var driver = new webDriver
					.Builder()
					//.usingServer(args.isMobile ? mainConfig.host.mobile : '')
					//.withCapabilities({ browserName: mainConfig.browserName })
					.withCapabilities({ browserName: 'chrome' })
					.build();

			//driver.manage().window().setSize(mainConfig.screen.width, mainConfig.screen.height);

			return driver;

		},
		getTest: function () {

			return fs.readdirSync( path.resolve(startPath, mainCfg.folder.test) );


		},
		getStartPath: function () {
			return startPath;
		},

		getArguments: function () {
			var data = {};
			process.argv.forEach(function (val) {
				var arr = val.split('=');
				data[arr[0]] = arr[1] || this.inKeyList(arr[0]);
			}, this);

			return data;
		}


	};

	module.exports = util;

}());
